### PR TITLE
(IMAGES-414) Add Win-2012R2 Core Packer Template

### PR DESCRIPTION
**Pull request by jcoconnor**
This is a copy of the Windows 2012r2 template with:
1. Replace SERVERSTANDARD with SERVERCORE
2. Update .Net to 4.6.1
3. Update slipstream with Core image updates.
4. Add workaround to get boxstarter to resume after reboot.

This PR also contains commits for the following issues:
1. IMAGES-576 - Switch to latest powershell module
2. IMAGES-578 - Update to .Net 4.7
3. IMAGES-553 - Fix to support packer upgrade
4. IMAGES-515 - Update WMF5 Templates (2008r2/2012r2) to WMF 5.1
5. IMAGES-573 - Add Win-2016 Core Template

### CODE REVIEW

File: scripts/windows/clean-disk-dism.ps1 [hunk 1]
```diff
@@ -35,6 +35,9 @@ ElseIf ($WindowsVersion -like $WindowsServer2012) {
 If ($WindowsVersion -like $WindowsServer2008) {
   Write-Host "Skipping CleanMgr for Windows 2008"
 }
+ElseIf ( $WindowsServerCore ) {
+  Write-Host "Skipping Clean-Mgr as GUI not installed (CORE Installation)."
```
Comment (glennsarti) [inline scripts/windows/clean-disk-dism.ps1]: nit: It's Core not CORE

Comment (jcoconnor) [inline scripts/windows/clean-disk-dism.ps1]: Nit fixed.

@glennsarti I concur on the Copy/Pasta.
Now that we are back not he same team, maybe we can collaborate to reduce it.
First up is to use xslt to simplify the ```unattend.xml```

I've some idea's for a variables file

**Comment by jcoconnor** [comment]
This is ready for merging pending review.
It builds and runs under the packer acceptance testing with 3 test failures.
1. tests/catalog_with_binary_data.rb error
2. tests/environment/use_enc_environment_for_files.rb
3. tests/reports/cached_catalog_status_in_report.rb

The first failure is a known issue (see PUP-7521).
The other two errors look to be related to the Ruby environment rather than the actual image.

Acceptance Test log is available at: https://jenkins-imaging.delivery.puppetlabs.net/view/Windows/job/imaging_windows-2012r2-core_x86_64_vmware_vsphere.cygwin_acceptance_packer/7/consoleFull

**Comment by jcoconnor** [comment]
This has passed acceptance testing on ```win-2012r2-core``` & ```win-2016-core```
It has also passed testing on all platforms with the exception of ```win-10``` (both 32 & 64).
The Win-10 issue is going to be dealt with in a separate ticket as it appears to be related to a recent Windows Update and is not connected with this PR (see [IMAGES-596](https://tickets.puppetlabs.com/browse/IMAGES-596))

This PR is ready for merge.
/cc @glennsarti @McdonaldSeanp @jpogran

**Comment by jcoconnor** [comment]
Acceptance test run for ```win-2016-core```: https://jenkins-imaging.delivery.puppetlabs.net/view/Windows/job/imaging_windows-2016-core_x86_64_vmware_vsphere.cygwin_acceptance_packer/1/console

Acceptance test run for ```win-2012r2-core```: https://jenkins-imaging.delivery.puppetlabs.net/view/Windows/job/imaging_windows-2012r2-core_x86_64_vmware_vsphere.cygwin_acceptance_packer/10/console

Note the test ```tests/catalog_with_binary_data.rb``` failure can be ignored as this is an issue with the test itself that has been fixed but not yet promoted to ```master```

File: scripts/windows/generate-slipstream.ps1 [hunk 2]
```diff
@@ -98,10 +98,15 @@ if ($WindowsVersion -like $WindowsServer2008R2 ) {
   # Windows 2008R2/Win-7 - just set registry keys for cleanmgr utility
   Write-Host "Skipping Cleanup"
 }
-ElseIf ($WindowsVersion -like $WindowsServer2012 -or $WindowsVersion -like $WindowsServer2008) {
+ElseIf ($WindowsVersion -like $WindowsServer2012R2 ) {
```
Comment (jpogran) [inline scripts/windows/generate-slipstream.ps1]: This if-else could be simplified with a switch statement

**Comment by jpogran** [comment]
You sure do a lot in one PR, other than powershell style suggestions you are good to go

File: scripts/windows/generate-slipstream.ps1 (hunk 2 above)
Comment (jcoconnor) [inline scripts/windows/generate-slipstream.ps1]: Agreed - I'm going to hold this over for the next PR as I have several other enhancements in the pipeline and think I have changed rather much already.

**Pull request merged**